src/NuGet.Core/NuGet.LibraryModel/LibraryDependency.cs

```diff
@@ -33,6 +33,11 @@ public class LibraryDependency : IEquatable<LibraryDependency>
         /// </summary>
         public bool VersionCentrallyManaged { get; set; }
 
+        /// <summary>
+        /// Information regarding if the dependency is direct or transitive.  
+        /// </summary>
+        public LibraryDependencyReferenceType ReferenceType { get; set; } = LibraryDependencyReferenceType.Direct;
```

Comment:
Feels like this adds duplication. 
TargetFrameworkInformation owns LibraryDependency and it owns CentralPackageVersions.
We explicitly wanted CentralPackageVersions to not be a part of LibraryDependency because, but now we are putting it back.

Reply:
I agree that it feels like duplication. But when I was looking more closely (and debugging), I think it's needed with the current implementation, so it can detect the difference between direct dependencies and transitive dependencies (direct dependencies do not need to be added at the end of the graph walk, like the transitive dependencies do). But if I'm wrong and it can be removed, that'd be great.

Reply:
Discussed on call.
My comments were stylistic and we have a consensus that it's ok to stay with the current implementation. There's not a great enough reason to change it.